src/jiti.ts

```diff
@@ -223,7 +223,20 @@ export default function createJITI (_filename: string, opts: JITIOptions = {}, p
 
     // Transpile
     const isTypescript = ext === '.ts'
-    const isNativeModule = ext === '.mjs'
+    let isNativeModule = ext === '.mjs'
+    let path = filename
+    if (ext === '.js') {
+      while (!isNativeModule && path && path !== '.' && path !== '/') {
```

Comment:
Maybe we can extract this function to utils and also use pkg-types for finding closest package.json?

Reply:
`pkg-types` would have been my first choice but it's async

Reply:
Shall we introduce a `sync` method there? It might be useful for edge cases like this and scripts.